packages/flutter/test/widgets/platform_menu_bar_test.dart

```diff
@@ -315,7 +317,7 @@ List<PlatformMenuItem> createTestMenus({
           members: <PlatformMenuItem>[
             PlatformMenuItem(
               label: subMenu1[0],
-              onSelected: onActivate != null ? () => onActivate(subMenu1[0]) : null,
+              onSelectedIntent: onActivateIntent?.call(subMenu1[0]),
```

Comment:
I'm a little surprised this doesn't cause the existing test to fail, but maybe it ends up producing the same thing? I'm surprised, because this change prevents the test from ever testing the `onActivate` case, since it's never used.

Anyhow, I think there need to be two tests: one that tests `onSelected`, and one that tests `onSelectedIntent`.  In this test menu creation function, you can just put back the line you deleted, and send in the nullable intent that came into the function.

While you're at it, can you rename the "onActivate*" arguments to this function to "onSelected*", since it should have been called that anyhow (it got missed in a name change).

Reply:
> I'm a little surprised this doesn't cause the existing test to fail, but maybe it ends up producing the same thing?

My understanding is yes, it just produces the same result. Looking at the test, it only has [two calls to `expect`]:

1. Expect that `Menu.setMenus` was called
2. Expect that the serialized version of the menu structure matches a hardcoded expected value

The serialized version of the menu items doesn't contain any information about whether a callback or intent was provided, only whether the item is enabled or disabled, so specifying either one satisfies the test.

>  I'm surprised, because this change prevents the test from ever testing the onActivate case, since it's never used.

As far as I can tell, there are currently no tests that actually test activating menu items. There is code [earlier in the test suite][2] that records which items have their callback invoked, but none of the tests ever actually select a menu item or `expect` that any items were selected. All that `onActivate` recording is just dead code right now.

> Anyhow, I think there need to be two tests: one that tests onSelected, and one that tests onSelectedIntent. In this test menu creation function, you can just put back the line you deleted, and send in the nullable intent that came into the function.
> While you're at it, can you rename the "onActivate*" arguments to this function to "onSelected*", since it should have been called that anyhow (it got missed in a name change).

SGTM and SGTM

[1]: https://github.com/jmatth/flutter/blob/c2aae6d83b8f437ee4cc76f6b7e6e23198ba0dba/packages/flutter/test/widgets/platform_menu_bar_test.dart#L72-L161
[2]: https://github.com/jmatth/flutter/blob/c2aae6d83b8f437ee4cc76f6b7e6e23198ba0dba/packages/flutter/test/widgets/platform_menu_bar_test.dart#L18-L32

Reply:
Thanks for pointing that out.  I'll have to add some more tests, I can tell.